src/Entity/BannerSelection/TrafficImpressionThreshold.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\BannerServer\Entity\BannerSelection;
+
+/**
+ * @license GNU GPL v2+
+ */
+class TrafficImpressionThreshold implements ImpressionThresholdInterface {
+
+	private $trafficLimit;
+	private $seasonalLimit;
+
+	public function __construct( float $trafficLimit, int $seasonalLimit ) {
+		$this->trafficLimit = $trafficLimit;
+		$this->seasonalLimit = $seasonalLimit;
+	}
+
+	public function getIsOverThreshold( int $visitorTotalImpressions ): bool {
+		return $this->isTrafficLimited() || $this->isImpressionLimitReached( $visitorTotalImpressions );
+	}
+
+	private function isTrafficLimited(): bool {
+		return (random_int( 0, 1 ) / 100) <= $this->trafficLimit;
```

Comment:
`random_int (0, 1)` will either create 0 or 1, I think you meant `random_int (0, 100)` . This is why you should write test-driven!

Speaking of test driven, in a unit test you can put the function `random_int` into the `WMDE\BannerServer\Entity\BannerSelection` namespace, to control the randomness, see https://stackoverflow.com/questions/18741147/how-to-unit-test-function-with-some-randomness